app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -439,45 +442,48 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     /**
      * Parses links from HTML string, and makes the links clickable in the specified TextView.<br>
-     * Uses {@link #makeLinkClickable(SpannableStringBuilder, URLSpan)}.
+     * Uses {@link //#makeLinkClickable(url,start,end,strBuilder)}.
      * @see <a href="https://stackoverflow.com/questions/12418279/android-textview-with-clickable-links-how-to-capture-clicks">Source</a>
      */
-    private void setTextViewHTML(TextView text, String html)
-    {
+
+    private void setTextViewHTML(TextView text, String html){
         CharSequence sequence = Html.fromHtml(html);
-        SpannableStringBuilder strBuilder = new SpannableStringBuilder(sequence);
+        SpannableString strBuilder = new SpannableString(sequence);
         URLSpan[] urls = strBuilder.getSpans(0, sequence.length(), URLSpan.class);
         for (URLSpan span : urls) {
-            makeLinkClickable(strBuilder, span);
+
+            int start = strBuilder.getSpanStart(span);
+            int end = strBuilder.getSpanEnd(span);
+            String url = span.getURL();
+
+            makeLinkClickable(url,start,end,strBuilder);
         }
         text.setText(strBuilder);
         text.setMovementMethod(LinkMovementMethod.getInstance());
     }
 
+
     /**
-     * Sets onClick handler to launch browser for the specified URLSpan.
-     * @see <a href="https://stackoverflow.com/questions/12418279/android-textview-with-clickable-links-how-to-capture-clicks">Source</a>
+     * URLSpan class extends ClickableSpan implements ParcelableSpan hance avoiding the memeory leak
+     *URLSpan  open the {@param url}, by launching an an Activity with an Intent#ACTION_VIEW intent.
+     * @see <a https://developer.android.com/reference/android/text/style/URLSpan">Source</a>
+
+     * @param  url        span URL giving to be clicked on
+     * @param  start      span start
+     * @param  end        span send
+     * @param  strBuilder span send
      */
-    private void makeLinkClickable(SpannableStringBuilder strBuilder, final URLSpan span)
-    {
-        int start = strBuilder.getSpanStart(span);
-        int end = strBuilder.getSpanEnd(span);
-        int flags = strBuilder.getSpanFlags(span);
-        ClickableSpan clickable = new ClickableSpan() {
-            public void onClick(View view) {
-                // Handle hyperlink click
-                String hyperLink = span.getURL();
-                launchBrowser(hyperLink);
-            }
-        };
-        strBuilder.setSpan(clickable, start, end, flags);
-        strBuilder.removeSpan(span);
-    }
 
-    private void launchBrowser(String hyperLink) {
-        Utils.handleWebUrl(this, Uri.parse(hyperLink));
+
+    private  final void makeLinkClickable(final String url, int start, int end, SpannableString  strBuilder) {
+
+        strBuilder.setSpan(new URLSpan(url),start, end, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+
     }
 
+
```

Comment:
Please remove spaces

Reply:
I reduced amount of code in the function above should is just extend the brackets

Reply:
Sorry I couldn't understand.

Reply:
The old cod was having many lines of codes and the new code is having less

Reply:
I have tried to fix it can you check it out now

Reply:
Can you please check lines 472 and 473, they don't make sense to me. Comments/Javadocs should be understandable. And can you please remove the lines 480, 481, 482. No space needed there.

Reply:
I have tried to work on this